### PR TITLE
Implement a TimeDisplay object

### PR DESCRIPTION
**Pull request by abraker95**

### CODE REVIEW

File: osu.Game/Graphics/TimeDisplay.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Graphics.TimeDisplay
+{
+    public class TimeDisplay : Drawable
+    {
+        private SpriteText timeText;
+        private bool _24Hour;
```
Comment (peppy) [inline osu.Game/Graphics/TimeDisplay.cs]: storing local for no reason

File: osu.Game/Graphics/TimeDisplay.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        protected override void Update()
+        {
+            if (_24Hour)
+                timeText.Text = DateTime.Now.ToString("HH:mm:ss");
+            else
+                timeText.Text = DateTime.Now.ToString("h:mm:ss tt");
+        }
+
+        public bool millitaryTime
```
Comment (peppy) [inline osu.Game/Graphics/TimeDisplay.cs]: `public bool MilitaryTime { get; set; }`

- millitary is incorrect spelling
- use auto-property rather than manually creating private storage.
- capital first letter for public variables

File: osu.Game/Graphics/TimeDisplay.cs (hunk 1 above)
Comment (abraker95) [inline osu.Game/Graphics/TimeDisplay.cs]: Which one?

timeText is needed to set the display text and _24Hour is needed to keep track of whether to display the 24 hour format or not.

Comment (peppy) [inline osu.Game/Graphics/TimeDisplay.cs]: the line i commented on

**Comment by peppy** [comment]
This shouldn't be in the Graphics namespace. Where's it going to be used? put it there.

honestly probably not even worth merging this in for now; my comments are just there to help you learn coding practices

**Comment by abraker95** [comment]
I wasn't sure under what namespace to put it. I chose Graphics since it looks like a graphics object to me.

**Comment by SirCmpwn** [comment]
:-1: to "MilitaryTime". Just have the user provide a format? This should be closed if it's not going to be used, though.

**Comment by abraker95** [comment]
You mean have the use provide something like "hh:mm:ss" in the options?

File: osu.Game/Graphics/TimeDisplay.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        private SpriteText timeText;
+        private string format;
+        private float textSize;
+        private Vector2 textPos;
+
+        public override void Load()
+        {
+            base.Load();
+            Children = new Drawable[]
```
Comment (peppy) [inline osu.Game/Graphics/TimeDisplay.cs]: should be in ctor now.

File: osu.Game/Graphics/TimeDisplay.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        public float TextSize
+        {
+            set
+            {
+                textSize = value;
+            }
+        }
+
+        public Vector2 TextPosition
```
Comment (peppy) [inline osu.Game/Graphics/TimeDisplay.cs]: why is this needed?

Comment (abraker95) [inline osu.Game/Graphics/TimeDisplay.cs]: I wanted to allow to adjust the text size depending on the application. Should I just make the SpriteText public then or is there a way for it to adjust automatically based on the container size?

**Comment by peppy** [comment]
closing this one for now
